<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from kuka-model.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="kuka-model" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!-- Prefix for joints and links -->
  <!-- KUKA software joint limits -5 degrees -->
  <!-- Axis a2 gedit and a3 offset (wrt KUKA axis angles) not includied here (view joints definitions)-->
  <!-- Alpha value for KUKA Shadow -->
  <!-- Joints -->
  <joint name="a1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.35"/>
    <parent link="base"/>
    <child link="link_1"/>
    <axis xyz="0 0 -1"/>
    <limit effort="0" lower="-3.141594" upper="3.141594" velocity="2.6529"/>
  </joint>
  <joint name="a2" type="revolute">
    <origin rpy="0 0 0" xyz="0.3 -0.068 0.315"/>
    <parent link="link_1"/>
    <child link="link_2"/>
    <axis xyz="0 1 0"/>
    <!-- Offset de -90 -->
    <limit effort="0" lower="-0.698132" upper="1.221731" velocity="2.6529"/>
  </joint>
  <joint name="a3" type="revolute">
    <origin rpy="0 0 0" xyz="0 -0.011 0.65"/>
    <parent link="link_2"/>
    <child link="link_3"/>
    <axis xyz="0 0.99939 -0.034899"/>
    <!-- Offset +90 -->
    <limit effort="0" lower="-3.5779265" upper="1.1344645" velocity="2.6529"/>
  </joint>
  <joint name="a4" type="revolute">
    <origin rpy="0 0 0" xyz="0.358 0.08636 0.15208"/>
    <parent link="link_3"/>
    <child link="link_4"/>
    <axis xyz="-1 0 0"/>
    <limit effort="0" lower="-6.0213885" upper="6.0213885" velocity="4.3633"/>
  </joint>
  <joint name="a5" type="revolute">
    <origin rpy="0 0 0" xyz="0.242 0.020238 -0.00070671"/>
    <parent link="link_4"/>
    <child link="link_5"/>
    <axis xyz="0 1 0"/>
    <limit effort="0" lower="-1.919863" upper="1.396264" velocity="6.2308"/>
  </joint>
  <joint name="a6" type="revolute">
    <origin rpy="0 0 0" xyz="0.1125 -0.02025 0"/>
    <parent link="link_5"/>
    <child link="link_6"/>
    <axis xyz="-1 0 0"/>
    <limit effort="0" lower="-6.0213885" upper="6.0213885" velocity="11.519"/>
  </joint>
  <!-- Links -->
  <link name="base">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/visual/base_link.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.2 0.5"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/collision/base_link.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="link_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.11674 0.022325 0.17065"/>
      <mass value="33.045"/>
      <inertia ixx="0.96011" ixy="0.14584" ixz="0.51813" iyy="1.5289" iyz="0.11532" izz="1.3182"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/visual/link_1.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.94 0.590 0.08 0.5"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/collision/link_1.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="link_2">
    <inertial>
      <origin rpy="0 0 0" xyz="-5.1658E-05 -0.076847 0.27349"/>
      <mass value="14.509"/>
      <inertia ixx="0.64764" ixy="6.686E-05" ixz="8.7057E-06" iyy="0.64831" iyz="-0.0047" izz="0.052771"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/visual/link_2.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.94 0.590 0.08 0.5"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/collision/link_2.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="link_3">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0026683 0.11641 0.087839"/>
      <mass value="13.751"/>
      <inertia ixx="0.15848" ixy="-0.021262" ixz="0.002384" iyy="0.32598" iyz="-0.0032541" izz="0.28494"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/visual/link_3.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.94 0.590 0.08 0.5"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/collision/link_3.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="link_4">
    <inertial>
      <origin rpy="0 0 0" xyz="0.11605 0.011227 -0.0003921"/>
      <mass value="2.1033"/>
      <inertia ixx="0.0040841" ixy="0.0015385" ixz="-5.3731E-05" iyy="0.012206" iyz="-5.7395E-05" izz="0.013848"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/visual/link_4.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.94 0.590 0.08 0.5"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/collision/link_4.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="link_5">
    <inertial>
      <origin rpy="0 0 0" xyz="0.023271 -0.036872 4.3221E-10"/>
      <mass value="0.64815"/>
      <inertia ixx="0.00060616" ixy="0.00023977" ixz="4.1189E-12" iyy="0.0012368" iyz="-1.4095E-13" izz="0.0013528"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/visual/link_5.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.94 0.590 0.08 0.5"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/collision/link_5.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="link_6">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0065821 3.3266E-05 -0.00027413"/>
      <mass value="0.023044"/>
      <inertia ixx="1.0462E-05" ixy="1.3009E-09" ixz="-1.072E-08" iyy="5.6691E-06" iyz="-3.9092E-08" izz="5.3518E-06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/visual/link_6.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.74 0.74 0.74 0.5"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/collision/link_6.stl"/>
      </geometry>
    </collision>
  </link>
  <!-- Include tool -->
  <link name="kinect_base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0092379 3.5234E-05 -0.10469"/>
      <mass value="0.7647"/>
      <inertia ixx="0.0042452" ixy="2.9964E-07" ixz="-6.0788E-05" iyy="0.0017558" iyz="-1.3174E-06" izz="0.0029547"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/tool/visual/kinect_gripper.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.4 0.4 0.4 0.5"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_model/meshes/tool/collision/kinect_gripper.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="depth_sensor"/>
  <joint name="kinect_base" type="fixed">
    <origin rpy="0 0 0" xyz="0.095 -0.02025 0"/>
    <parent link="link_5"/>
    <child link="kinect_base_link"/>
  </joint>
  <joint name="kinect_fixed" type="fixed">
    <origin rpy="0 1.5435 0" xyz="-0.0013956 0.011693 -0.14997"/>
    <parent link="kinect_base_link"/>
    <child link="depth_sensor"/>
  </joint>
</robot>

